xapian-applications/omega/query.cc

```diff
@@ -1940,6 +1955,19 @@ eval(const string &fmt, const vector<string> &param)
 		value = args[0];
 		url_prettify(value);
 		break;
+	    case CMD_qid: {
+		string qstr = args[0];
+		trim_string(qstr);
```

Comment:
> What is using this hashed query id supposed to accomplish?

Query ID basically provides a means to uniquely identify each query from clickstream log if we need to do so in the future. For now, it's being particularly helpful in preprocessing step which I'm currently onto, for obtaining the final clickstream log file.

> If it's to track a particular user's "search session", then trimming spaces doesn't seem necessary - users don't commonly repeat the exact same query but add or remove spaces at the start or end or it.

This was based on the idea that leading or trailing whitespaces in query string wouldn't make query anything more useful other than leading to some duplicate log entries so I thought it was best to just rid off them to keep log file neat.

Do you suggest using the transform command that you mentioned earlier `$transform{$query,^ *(.*?) *$,\1}` or this `$transform{ $query ,\s+, ,g}` for dealing with whitespaces?